src/Step/Deploy/InstallUpdate/ConfigUpdate/DbConnection.php

```diff
@@ -60,82 +61,272 @@ class DbConnection implements StepInterface
     /**
      * @var RelationshipConnectionFactory
      */
-    private $connectionFactory;
+    private $connectionDataFactory;
+
+    /**
+     * @var FlagManager
+     */
+    private $flagManager;
 
     /**
      * @param DeployInterface $stageConfig
-     * @param MergedConfig $mergedConfig
+     * @param DbConfig $dbConfig
      * @param ResourceConfig $resourceConfig
      * @param ConfigWriter $configWriter
      * @param ConfigReader $configReader
      * @param ConfigMerger $configMerger
-     * @param RelationshipConnectionFactory $connectionFactory
+     * @param RelationshipConnectionFactory $connectionDataFactory
      * @param LoggerInterface $logger
+     * @param FlagManager $flagManager
      */
     public function __construct(
         DeployInterface $stageConfig,
-        MergedConfig $mergedConfig,
+        DbConfig $dbConfig,
         ResourceConfig $resourceConfig,
         ConfigWriter $configWriter,
         ConfigReader $configReader,
         ConfigMerger $configMerger,
-        RelationshipConnectionFactory $connectionFactory,
-        LoggerInterface $logger
+        RelationshipConnectionFactory $connectionDataFactory,
+        LoggerInterface $logger,
+        FlagManager $flagManager
     ) {
         $this->stageConfig = $stageConfig;
-        $this->mergedConfig = $mergedConfig;
+        $this->dbConfig = $dbConfig;
         $this->resourceConfig = $resourceConfig;
         $this->configWriter = $configWriter;
         $this->configReader = $configReader;
         $this->configMerger = $configMerger;
         $this->logger = $logger;
-        $this->connectionFactory = $connectionFactory;
+        $this->connectionDataFactory = $connectionDataFactory;
+        $this->flagManager = $flagManager;
     }
 
     /**
      * @inheritdoc
```

Comment:
Please add a description. And describe when `FLAG_IGNORE_SPLIT_DB` is set.

Reply:
Done